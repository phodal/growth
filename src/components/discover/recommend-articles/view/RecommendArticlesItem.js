import React, { Component, PropTypes } from 'react';
import { Text, View, TouchableHighlight } from 'react-native';
import AppStyle from '../../../../theme/styles';
import Line from '../../../Line';

class RecommendArticlesItem extends Component {
  static componentName = 'RecommendArticlesItem';

  static propTypes = {
    content:
        PropTypes.shape(
          PropTypes.string.isRequired,
          PropTypes.arrayOf(PropTypes.shape(
            PropTypes.string.isRequired,
            ),
          )),
  };

  static defaultProps = {
    content: {},
  };

  constructor(props) {
    super(props);
    this.state = {
      name: this.props.content.name,
      rowData: Array.from(new Array(this.props.content.books.length))
        .map((val, index) => (this.props.content.books[index])) };
  }

  render() {
    const rows = this.state.rowData.map((val, index) => (
      <TouchableHighlight onPress={() => console.log('jupm book detail')} key={val.title.concat(index)}>
        <View style={{ backgroundColor: 'white' }}>
          <View style={AppStyle.projectListContentStyle}>
            <Text>{val.title}</Text>
          </View>
          <Line />
        </View>
      </TouchableHighlight>
     ));
    return (
      <View>
        <View style={AppStyle.projectListTitleStyle}>
          <Text style={{ color: '#999' }}>{this.state.name}</Text>
        </View>
        {rows}
      </View>
    );
  }
}
export default RecommendArticlesItem;

