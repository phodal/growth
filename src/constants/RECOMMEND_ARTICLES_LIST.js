import DOMAIN_NAME from './DOMAIN_NAME';
import HELPER_ARTICLES from './HELPER_ARTICLES';

const RECOMMEND_ARTICLES_LIST = [
  { name: DOMAIN_NAME.hello, books: HELPER_ARTICLES['zh-cn'].hello },
  { name: DOMAIN_NAME.env, books: HELPER_ARTICLES['zh-cn'].env },
  { name: DOMAIN_NAME.build, books: HELPER_ARTICLES['zh-cn'].build },
  { name: DOMAIN_NAME.front, books: HELPER_ARTICLES['zh-cn'].front },
  { name: DOMAIN_NAME.backend, books: HELPER_ARTICLES['zh-cn'].backend },
  { name: DOMAIN_NAME.before, books: HELPER_ARTICLES['zh-cn'].before },
  { name: DOMAIN_NAME.test, books: HELPER_ARTICLES['zh-cn'].test },
  { name: DOMAIN_NAME.coding, books: HELPER_ARTICLES['zh-cn'].coding },
  { name: DOMAIN_NAME.container, books: HELPER_ARTICLES['zh-cn'].container },
  { name: DOMAIN_NAME.server, books: HELPER_ARTICLES['zh-cn'].server },
  { name: DOMAIN_NAME.configurable, books: HELPER_ARTICLES['zh-cn'].configurable },
  { name: DOMAIN_NAME.analytics, books: HELPER_ARTICLES['zh-cn'].analytics },
  { name: DOMAIN_NAME.ux, books: HELPER_ARTICLES['zh-cn'].ux },
  { name: DOMAIN_NAME.ci, books: HELPER_ARTICLES['zh-cn'].ci },
  { name: DOMAIN_NAME.cd, books: HELPER_ARTICLES['zh-cn'].cd },
  { name: DOMAIN_NAME.legacy, books: HELPER_ARTICLES['zh-cn'].legacy },
  { name: DOMAIN_NAME.refactor, books: HELPER_ARTICLES['zh-cn'].refactor },
  { name: DOMAIN_NAME.retro, books: HELPER_ARTICLES['zh-cn'].retro },
  { name: DOMAIN_NAME.arch, books: HELPER_ARTICLES['zh-cn'].arch },
];
export default RECOMMEND_ARTICLES_LIST;
