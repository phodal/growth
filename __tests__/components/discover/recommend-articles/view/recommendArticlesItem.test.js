import 'react-native';
import React, {} from 'react';
import renderer from 'react-test-renderer';
import RecommendArticlesItem from '../../../../../src/components/discover/recommend-articles/view/RecommendArticlesItem';
import RECOMMEND_ARTICLES_LIST from '../../../../../src/constants/RECOMMEND_ARTICLES_LIST';

it('renders correctly', () => {
  const recommendArticlesItem = renderer.create(
    <RecommendArticlesItem content={RECOMMEND_ARTICLES_LIST[0]} />,
  );
  const recommendArticlesItemJson = recommendArticlesItem.toJSON();
  expect(recommendArticlesItemJson).toMatchSnapshot();
});
