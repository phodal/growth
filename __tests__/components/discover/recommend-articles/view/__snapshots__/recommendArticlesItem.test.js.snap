// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "rgb(245,245,245)",
        "height": 40,
        "justifyContent": "center",
        "paddingLeft": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#999",
        }
      }
    >
      hello, world
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          如何学好一门技术、语言
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(240, 240, 240, .9)",
            "height": 1,
          }
        }
      />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          HTML入门
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(240, 240, 240, .9)",
            "height": 1,
          }
        }
      />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          CSS入门
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(240, 240, 240, .9)",
            "height": 1,
          }
        }
      />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          JavaScript入门
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(240, 240, 240, .9)",
            "height": 1,
          }
        }
      />
    </View>
  </View>
</View>
`;
